Latest version with todos for tests

simona1 opened the pull request:
Will add the tests soon  and update the PR respectively - this has been addressed.

HamidAghdaee commented on src/Actions.js:
```diff
@@ -0,0 +1,118 @@
+type Dispatch = ({type: string}) => void;
+type GetState = () => State;
+
+function fakePromise(data, delay) {
```
what's the use for fake data? I thought you were connected to a live data source. please remove all references to a fake source of data.

HamidAghdaee commented on src/store.js:
```diff
@@ -0,0 +1,122 @@
+// @flow
```
please create a reducers folder and put the reducers in there. you will also need to export the initial reducer from there as well

HamidAghdaee commented on src/Toolbar.jsx:
```diff
@@ -1,33 +1,78 @@
+
 import './index.css';
+import store from './store';
+import Actions from './Actions';
```
actions should be in their own folder

HamidAghdaee commented on src/Toolbar.jsx:
```diff
@@ -1,33 +1,78 @@
+    if (channels.length === 0) {
+      store.dispatch(Actions.fetchChannels());
+    }
+    store.subscribe(() => this.forceUpdate());
```
all connections to the store need to be done through react-redux's connect function

HamidAghdaee commented on src/Toolbar.jsx:
```diff
@@ -1,33 +1,78 @@
+  componentWillMount() {
+    const channels = Object.keys(store.getState().channelData);
+    if (channels.length === 0) {
+      store.dispatch(Actions.fetchChannels());
```
all dispatches to the store need to be done through react-redux's connect function. The only place you should ever need to be importing the store is in src/index.js. nowhere else

HamidAghdaee commented on src/index.js:
```diff
@@ -10,3 +14,15 @@ ReactDOM.render(
   document.getElementById('root'),
 );
 registerServiceWorker();
+
```
The only time you're ever going to import store in your code should be in index.js and only to provide it in the provider component.

HamidAghdaee replied:
Please create another file for processing messages on a socket connection,

HamidAghdaee commented on src/Actions.js:
```diff
@@ -0,0 +1,118 @@
+// @flow
```
don't use a default export for all actions. just individual exports for each function

simona1 commented:
Code has been refactored to address comments. Passes lint and flow checks and tests.

HamidAghdaee commented on src/Components/MessageList.jsx:
```diff
@@ -0,0 +1,67 @@
+    let { messages } = this.props;
+    if (!messages) {
+      if (selectedChannel) {
+        setTimeout(
```
The render function should just be a pure function that returns React elements (or JSX). This code should go either in lifecycle methods or another utils function

HamidAghdaee commented on src/FlowTypes/Types.js:
```diff
@@ -0,0 +1,21 @@
+export type Id = mixed;
```
Please rename this to index.js and everywhere else, where the types need to be imported, use `import * from '/FlowTypes'` or something like that. The same goes for the Constants folder as well as the Utils folder

HamidAghdaee commented:
nice job @simona1. Please see my comments. Also your tests fail for me. see: 
<img width="1145" alt="screen shot 2017-08-23 at 10 36 40 am" src="https://user-images.githubusercontent.com/15915015/29629946-77eb7206-87ef-11e7-9619-5bab0c10fcb7.png">

Your tests should not depend on the network connection at all. See: 

https://github.com/gSchool/react-storybook/blob/cart-20-redux-alternative/src/actions/actions.test.js

simona1 commented:
I've addressed most of the comments - and added TODOs for the portions of the code that will need to be refactored for dashboard integration as we discussed. Code passes lint and flow checks. 

Tests pass on my computer with the mock server - see screenshot. Working on refactoring them as per your comment. Thanks so much!
![screen shot 2017-08-23 at 3 02 52 pm](https://user-images.githubusercontent.com/7232251/29640608-7ee1de0e-8814-11e7-849a-b60aacaea162.png)

HamidAghdaee commented:
@simona1 I am still seeing the same test fails. I'm fairly certain that it is because it is making a real fetch request, which it shouldn't be doing. See the file I referenced above. You should be using thunk with extra argument and pass the API into the thunk there, this way you'll have the ability to mock the api call.
